core/base-service/auth-helper.js

```diff
@@ -1,14 +1,21 @@
 'use strict'
 
 class AuthHelper {
-  constructor({ userKey, passKey, isRequired = false }, privateConfig) {
+  constructor(
+    { userKey, passKey, isRequired = false, useEmptyStringForUser = false },
+    privateConfig
+  ) {
     if (!userKey && !passKey) {
       throw Error('Expected userKey or passKey to be set')
     }
 
     this._userKey = userKey
     this._passKey = passKey
-    this.user = userKey ? privateConfig[userKey] : undefined
+    if (userKey) {
+      this.user = privateConfig[userKey]
+    } else {
+      this.user = useEmptyStringForUser ? '' : undefined
+    }
```

Comment:
I figured this would be a bit more readable than nested ternary statements, but I don't feel terribly strongly about it if anyone has a strong preference for the other